### PR TITLE
Roe 2398 individual bo changes

### PR DESCRIPTION
**Pull request by markpit**
### JIRA link
https://companieshouse.atlassian.net/browse/ROE-2398

### Change description
Add url params to the BO Individual page to create a 'happy path' journey through the service.
Screen validation will be worked on under story https://companieshouse.atlassian.net/browse/ROE-2409

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: test/controllers/beneficial.owner.individual.controller.spec.ts [hunk 1]
```diff
@@ -178,7 +224,7 @@ describe("BENEFICIAL OWNER INDIVIDUAL controller", () => {
     });
 
     test(`redirect to the ${BENEFICIAL_OWNER_TYPE_PAGE} page after a successful post from ${BENEFICIAL_OWNER_INDIVIDUAL_PAGE} page with service address data`, async () => {
-      mockPrepareData.mockImplementationOnce( () => BENEFICIAL_OWNER_INDIVIDUAL_OBJECT_MOCK_WITH_SERVICE_ADDRESS_NO );
+      mockPrepareData.mockReturnValueOnce( { ...BENEFICIAL_OWNER_INDIVIDUAL_OBJECT_MOCK_WITH_SERVICE_ADDRESS_NO } );
```
Comment (markpit) [inline test/controllers/beneficial.owner.individual.controller.spec.ts:227]: One of the new 'with params' tests was passing when run on its own but failed when run with other tests. I narrowed down the issue to the mock data being returned by the mockPrepareData being the same in the test that failed as some others and one of the other tests was modifying the mock object.
To fix it I changed all places that returned this mock data to return a clone of the object instead, so any changes made to that object would not be carried into other tests.
The clone is done using the spread syntax to copy the fields into the inside of a new object - eg
{ ...myObject }

File: src/controllers/beneficial.owner.type.controller.ts [hunk 2]
```diff
@@ -36,7 +50,7 @@ export const get = (req: Request, res: Response, next: NextFunction) => {
 export const post = (req: Request, res: Response) => {
   logger.debugRequest(req, `${req.method} ${req.route.path}`);
 
-  return res.redirect(getNextPage(req.body[BeneficialOwnerTypeKey]));
+  return res.redirect(getNextPage(req, req.body[BeneficialOwnerTypeKey]));
```
[the remark this answers is not in the document]
Comment (markpit) [reply src/controllers/beneficial.owner.type.controller.ts:53]: yes should be able to, i can update that

[the remark this answers is not in the document]
Comment (markpit) [reply src/controllers/beneficial.owner.type.controller.ts]: updated

File: src/controllers/beneficial.owner.type.controller.ts [hunk 3]
```diff
@@ -21,6 +22,19 @@ export const get = (req: Request, res: Response, next: NextFunction) => {
 
     logger.infoRequest(req, `${config.BENEFICIAL_OWNER_TYPE_PAGE} requiresTrusts=${requiresTrusts}`);
 
+    if (isActiveFeature(config.FEATURE_FLAG_ENABLE_REDIS_REMOVAL)) {
+      return res.render(config.BENEFICIAL_OWNER_TYPE_PAGE, {
+        FEATURE_FLAG_ENABLE_REDIS_REMOVAL: true,
```
Comment (mwestacott) [inline src/controllers/beneficial.owner.type.controller.ts:27]: I think we should add some comments in around this, to explain why it's not being done in the usual way (to facilitate testing). I can do that now.

Comment (mwestacott) [reply src/controllers/beneficial.owner.type.controller.ts:29]: Done.

**Review by dlloyd1-ch — APPROVED** [review]

**Pull request merged**